Use ServiceTrafficDistribution to make Services topology-aware when runtime Kubernetes >= 1.31 (#11178)
gardener/gardener: docs/operations/topology_aware_routing.md

By default, with kube-proxy running in `iptables` mode, traffic is distributed randomly across all endpoints, regardless of where it originates from. In a cluster with 3 zones, traffic is more likely to go to another zone than to stay in the current zone.
With the topology-aware routing feature, kube-proxy filters the endpoints it routes to based on the hints in the EndpointSlice resource. In most of the cases, kube-proxy will prefer the endpoint(s) in the same zone. For more details, see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#implementation-kube-proxy).

### How `ServiceTrafficDistribution` works

We reported the drawbacks related to the `TopologyAwareHints` feature in [kubernetes/kubernetes#113731](https://github.com/kubernetes/kubernetes/issues/113731). As result, the Kubernetes community implemented the `ServiceTrafficDistribution` feature.

The `ServiceTrafficDistribution` allows expressing preferences for how traffic should be routed to Service endpoints. For more details, see [upstream documentation](https://kubernetes.io/docs/reference/networking/virtual-ips/#traffic-distribution) of the feature.

> The `PreferClose` strategy with kube-proxy of `ServiceTrafficDistribution` allows traffic to be routed to Service endpoints in topology-aware and predictable manner.

Reviewer: ```suggestion
The `PreferClose` option allows traffic to be routed to Service endpoints in topology-aware and predictable manner.
```
